plugins/utils/refactoring.nix

```diff
@@ -6,140 +6,163 @@
   ...
 }:
 with lib;
-{
-  options.plugins.refactoring = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption "refactoring.nvim";
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "refactoring";
+  originalName = "refactoring.nvim";
+  defaultPackage = pkgs.vimPlugins.refactoring-nvim;
+
+  maintainers = [ helpers.maintainers.MattSturgeon ];
+
+  # TODO: introduced 2024-05-24, remove on 2024-08-24
+  optionsRenamedToSettings = [
+    "promptFuncReturnType"
+    "promptFuncParamType"
+    "printVarStatements"
+    "printfStatements"
+    "extractVarStatements"
+  ];
 
-    package = helpers.mkPluginPackageOption "refactoring.nvim" pkgs.vimPlugins.refactoring-nvim;
+  extraOptions = {
+    enableTelescope = mkEnableOption "telescope integration";
+  };
+
+  extraConfig = cfg: {
+    assertions = [
+      {
+        assertion = cfg.enableTelescope -> config.plugins.telescope.enable;
+        message = ''
+          Nixvim: You have enabled the `telescope` integration with refactoring-nvim.
+          However, you have not enabled the `telescope` plugin itself (`plugins.telescope.enable = true`).
+        '';
+      }
+    ];
 
-    promptFuncReturnType =
+    plugins.telescope.enabledExtensions = mkIf cfg.enableTelescope [ "refactoring" ];
+  };
+
+  settingsOptions = {
+    prompt_func_return_type =
       helpers.defaultNullOpts.mkNullable (with types; attrsOf bool)
```

Comment:
We should use mkAttrsOf if possible (this enables users to user raw lua)

Reply:
:+1: